docs/source/core_specification/schema_types.rst

```diff
@@ -1,3 +1,282 @@
+   like Entity, are not intuitive.
+
+.. code::
+
+    schema {
+        query: RootSchemaQuery
+    }
+
+    type RootSchemaQuery {
+        Animal: [Animal]
+        Entity: [Entity]
+        Food: [Food]
+        Species: [Species]
+        Toy: [Toy]
+    }
+
+    directive @filter(op_name: String!, value: [String!]) on FIELD | INLINE_FRAGMENT
+
+    directive @tag(tag_name: String!) on FIELD
+
+    directive @output(out_name: String!) on FIELD
+
+    directive @output_source on FIELD
+
+    directive @optional on FIELD
+
+    directive @recurse(depth: Int!) on FIELD
+
+    directive @fold on FIELD
+
+    scalar Date
+
+    scalar DateTime
+
+    scalar Decimal
+
+    type Animal implements Entity {
+        name: String
+        alias: [String]
+        birthday: Date
+        net_worth: Decimal
+        in_Animal_ParentOf: [Animal]
+        out_Animal_ParentOf: [Animal]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        out_Animal_OfSpecies: [Species]
+        out_Animal_PlaysWith: [Toy]
+    }
+
+    type Food implements Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+    }
+
+    type Species implements Entity {
+        name: String
+        alias: [String]
+        description: String
+        in_Animal_OfSpecies: [Animal]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+        out_Species_Eats: [Union__Food__Species]
+    }
+
+    type Toy {
+        name: String
+        in_Animal_PlaysWith: [Animal]
+    }
+
+    interface Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+    }
+
+    union Union__Food__Species = Food | Species
+
+
+.. note::
+
+    A GraphQL schema can be serialized with the :code:`print_schema` function from the
+    :code:`graphql.utils.schema_printer` module.
+
+
+Objects and Fields
+--------------------------
+
+The core components of a GraphQL schema are GraphQL object types. They conceptually represent the
+concrete vertices in the underlying database. For relational databases, we think of the tables as
+the concrete vertices.
+
+Lets go over a toy example of a GraphQL object type:
+
+.. code::
+
+    type Toy {
+        name: String
+        in_Animal_PlaysWith: [Animal]
+    }
+
+Here are some of the details:
+
+- :code:`name` is a **property field** that represents concretes data.
+- :code:`in_Animal_PlaysWith` is a **vertex field** representing an outbound edge.
+- :code:`String` is a built-in GraphQL scalar type.
+- :code:`[Animal]` is a GraphQL list type that represents an array of :code:`Animal`
+  objects.
+
+
+Directives
+----------
+
+In this section we'll go over how query directives are defined. For information on the available
```

Comment:
`In this section we'll go over how query directives are defined.` this sentence does not really convey any  new information.